sphinx/util/images.py

```diff
@@ -75,8 +75,13 @@ def guess_mimetype_for_stream(stream, default=None):
         return default
 
 
-def guess_mimetype(filename='', content=None, default=None):
+def guess_mimetype(filename=None, content=None, default=None):
     # type: (str, bytes, str) -> str
+    if filename is None:
+        warnings.warn('The first positional argument of guess_mimetype() will '
+                      'be required in Sphinx 4.0.',
+                      RemovedInSphinx40Warning, stacklevel=2)
```

Comment:
I guess this warning is mainly shown for developers who calls this with `content` parameter. And we've already emitted a warning for them below. So I wonder is this really helpful.

Reply:
I think no additional steps is needed. We'll modify it without any declaration. It only affect to `guess_mimetype(default='foo')` :-p

How about adding TODO comment to the warning for `content`? It will help us on removal.

Reply:
Done in the latest revision. Thanks!